engine/table/src/main/java/io/deephaven/engine/updategraph/impl/EventDrivenUpdateGraph.java

```diff
@@ -54,13 +54,23 @@ public int parallelismFactor() {
     @Override
     public void requestRefresh() {
         maybeStart();
-        // do the work to refresh everything, on this thread
-        isUpdateThread.set(true);
-        try (final SafeCloseable ignored = ExecutionContext.newBuilder().setUpdateGraph(this).build().open()) {
-            refreshAllTables();
-        } finally {
-            isUpdateThread.remove();
-        }
+        // Do the work to refresh everything, driven by this thread. Note that we acquire the lock "early" in order to
+        // avoid any inconsistencies w.r.t. assumptions about clock, lock, and update-thread state.
+        final long lockStartTimeNanos = System.nanoTime();
+        exclusiveLock().doLocked(() -> {
+            reportLockWaitNanos(System.nanoTime() - lockStartTimeNanos);
+            final boolean wasUpdateThread = isUpdateThread.get();
```

Comment:
If we were an update thread, we should be in refreshAllTables ... so it seems dangerous to permit this to be nested.  I'm not sure what "good" would come out of that and it seems like having completing cycles from the same thread would be bad.